torch/csrc/jit/codegen/cuda/lower_thread_predicate.h

```diff
@@ -48,9 +48,24 @@ class TORCH_CUDA_CU_API ThreadPredicateMap {
     ParallelTypeBitmap limited_types;
     // Parallel types where only one thread/block is enough.
     ParallelTypeBitmap redundant_types;
+    // Tracking use chain of redundant writes:
+    //  [Redundant use chain]
+    //  a parallel type is a `redundant_consumer_type` only
+    //    if all of its propagation use chains terminate with
+    //    a redundant write of this type.
+    //  A propagation use chain is currently either a reg-to-reg
+    //   chain for a shared mem tv, or a reg/smem-to-reg/smem chain
+    //   for a global tv.
+    // This is complementary information to `redundant_types`.
+    //  If a tensor view is redundantly written and not fully
+    //  redundantly used, see FusionRedundantPredSync3,
```

Comment:
I couldn't immediately get what "not fully redundantly used" means. Maybe just something like "unless redundantly used by all consumers".

Reply:
Updated. Thanks!